memory/eatmemory.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2016 IBM
+# Author: Santhosh G <santhog4@linux.vnet.ibm.com>
+
+import os
+from avocado import Test
+from avocado.utils import build
+from avocado.utils import memory
+from avocado.utils import process
+from avocado.utils import distro
+from avocado.utils import archive
+from avocado.utils.software_manager import SoftwareManager
+
+
+class eatmemory(Test):
+
+    def setUp(self):
+        sm = SoftwareManager()
+        detected_distro = distro.detect()
+        deps = ['gcc', 'make']
+        for package in deps:
+            if not sm.check_installed(package) and not sm.install(package):
+                self.error(package + ' is needed for the test to be run')
+        url = 'https://github.com/julman99/eatmemory/archive/master.zip'
+        tarball = self.fetch_asset("eatmemory.zip", locations=[url])
```

Comment:
Depending on the version of avocado (>=38), you can set the expire time for the asset file, so avocado will download it again if expired. Useful to refresh the file on your cache directory:

`tarball = self.fetch_asset("eatmemory.zip", locations=[url],expire='7d')`